cmd/readPipelineEnv.go

```diff
@@ -45,3 +83,56 @@ func runReadPipelineEnv() error {
 
 	return nil
 }
+
+func encrypt(secret, inBytes []byte) ([]byte, error) {
+	// use SHA256 as key
+	key := sha256.Sum256(secret)
+	block, err := aes.NewCipher(key[:])
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new cipher: %v", err)
+	}
+
+	// Make the cipher text a byte array of size BlockSize + the length of the message
+	cipherText := make([]byte, aes.BlockSize+len(inBytes))
+
+	// iv is the ciphertext up to the blocksize (16)
+	iv := cipherText[:aes.BlockSize]
+	if _, err = io.ReadFull(rand.Reader, iv); err != nil {
+		return nil, fmt.Errorf("failed to init iv: %v", err)
+	}
+
+	// Encrypt the data:
+	stream := cipher.NewCFBEncrypter(block, iv)
+	stream.XORKeyStream(cipherText[aes.BlockSize:], inBytes)
+
+	// Return string encoded in base64
+	return []byte(base64.StdEncoding.EncodeToString(cipherText)), err
+}
+
+// retrieve step metadata
+func readPipelineEnvMetadata() config.StepData {
+	var theMetaData = config.StepData{
+		Metadata: config.StepMetadata{
+			Name: "readPipelineEnvMetadata",
```

Comment:
Usually `Name` is a step name, not function name. Haven't dived into details, don't know if it is important.